lib/discovery-indexer/reader/purlxml_parser_strict.rb

```diff
@@ -142,12 +144,32 @@ def parse_dor_content_type
         dct
       end
 
-      # the @id attribute of resource/file elements that match the image type, including extension
+      # the @id attribute of resource/file elements that match the image, page, or thumb resource type, including extension
       # @return [Array<String>] filenames
       def parse_image_ids
         content_md = parse_content_metadata
         return nil if content_md.nil?
-        content_md.xpath('//resource[@type="page" or @type="image" or @type="thumb"]/file[@mimetype="image/jp2"]/@id').map(&:to_s)
+        ids = content_md.xpath('//resource[@type="page" or @type="image" or @type="thumb"]/file[@mimetype="image/jp2"]/@id').map(&:to_s)
+        return nil if ids.empty?
+        ids
+      end
+
+      # the druid and id attribute of resource/file and objectId and fileId of the
+      # resource/externalFile elements that match the image, page, or thumb resource type, including extension
+      # Also, prepends the corresponding druid and %2F specifically for Searchworks use
+      # @return [Array<String>] filenames
+      def parse_sw_image_ids
+        content_md = parse_content_metadata
+        return nil if content_md.nil?
+        ids = []
+        content_md.xpath('//resource[@type="page" or @type="image" or @type="thumb"]').each do |node|
+          ids = node.xpath('//file[@mimetype="image/jp2"]/@id').map{ |x| "#{@druid}%2F" + x }
+          node.xpath('//externalFile[@mimetype="image/jp2"]').each do |y|
+            ids << "#{y.attributes['objectId'].text.split(':').last}" + "%2F" + "#{y.attributes['fileId']}"
+          end
+        end
+        return nil if ids.empty?
+        ids
```

Comment:
3. I thought about that but I wanted to do as little refactoring as possible because it was fewer changes.
4. I want to keep things consistent for refactoring later.